src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -1228,6 +1232,9 @@ def __call__(
             if needs_upcasting:
                 self.upcast_vae()
                 latents = latents.to(next(iter(self.vae.post_quant_conv.parameters())).dtype)
+            elif torch.backends.mps.is_available() and latents.dtype != self.vae.dtype:
+                # some platforms (eg. apple mps) misbehave due to a pytorch bug, this is a workaround
```

Comment:
Would be also nice to include a link if there's any public report of this bug on the PyTorch GitHub repository.

Reply:
https://github.com/pytorch/pytorch/pull/99272 is involved, not sure if there are additional issues.